File: src/main/java/me/onebone/minecombat/task/TickTask.java
package me.onebone.minecombat.task;

import me.onebone.minecombat.MineCombat;

import cn.nukkit.scheduler.PluginTask;

public class TickTask extends PluginTask<MineCombat>{
	public TickTask(MineCombat plugin){
		super(plugin);
	}
	
	public void onRun(int currentTick){
		this.getOwner().onTick();
	}
}

File: src/main/kotlin/task/TickTask.kt
package me.onebone.minecombat.task

import cn.nukkit.scheduler.PluginTask
import me.onebone.minecombat.MineCombat

class TickTask(plugin: MineCombat, private val func: (Int) -> Unit): PluginTask<MineCombat>(plugin) {
	override fun onRun(currentTick: Int) {
		func(currentTick)
	}
}